source/game/system/map/MapdataPointInfo.hh

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+#include <egg/util/Stream.hh>
+
+#include <vector>
+
+namespace System {
+
+class MapdataPointInfo {
+public:
+    struct Point {
+        EGG::Vector3f pos;
+        u16 setting[2];
+    };
+
+    struct SData {
+        u16 pointCount;
+        u8 settings[2];
+        Point points[];
+    };
+    STATIC_ASSERT(sizeof(SData) == 0x4);
+
+    MapdataPointInfo(const SData *data);
+
+    void read(EGG::RamStream &stream);
+
+    [[nodiscard]] u16 pointCount() const;
+
+private:
+    const SData *m_rawData;
+    u16 m_pointCount;
+    std::array<u8, 2> m_settings;
+    std::vector<Point> m_points;
```

Comment:
A span would be more appropriate because there is never any more or less points than specified by the point count.